tasks/skynet/src/main/java/com/innowise/simulation/RobotWarSimulator.java

```diff
@@ -0,0 +1,74 @@
+package com.innowise.simulation;
+
+import com.innowise.service.Faction;
+import com.innowise.service.Factory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.CyclicBarrier;
+
+/**
+ * The RobotWarsSimulator class for running the robot wars simulation.
+ * It creates the threads for the factory and factions,
+ * starts them, and reports the final results.
+ */
+public class RobotWarSimulator {
+
+    private static final Logger LOGGER = LogManager.getLogger(RobotWarSimulator.class.getName());
+    private static final int SIMULATION_DAYS = 100;
```

Comment:
There's no explicit check that the simulation runs for exactly 100 days as required